### PR TITLE
Image: More cleanup and reduced code duplication

### PR DESCRIPTION
**Pull request by BlueCube3310**
Depends on #98084

This PR reduces code duplication in the Image class (mostly for mipmap generation), and optimizes detecting color channels.

### CODE REVIEW

File: core/io/image.cpp [hunk 1]
```diff
@@ -572,21 +572,18 @@ static bool _are_formats_compatible(Image::Format p_format0, Image::Format p_for
...
-		return;
-	}
 
-	if (p_new_format == format) {
+	if (data.size() == 0 || p_new_format == format) {
 		return;
 	}
 
+	ERR_FAIL_COND_MSG(Image::is_format_compressed(format) || Image::is_format_compressed(p_new_format),
+			"Cannot convert to <-> from compressed formats. Use compress() and decompress() instead.");
```
Comment (Calinou) [inline core/io/image.cpp:581]: Use an easier to understand form:

```suggestion
	ERR_FAIL_COND_MSG(Image::is_format_compressed(format) || Image::is_format_compressed(p_new_format),
			"Cannot convert to (or from) compressed formats. Use compress() and decompress() instead.");
```

**Review by Calinou — APPROVED** [review]
Code looks good to me.

This refactor will be helpful if we decide to [implement more mipmap](https://github.com/godotengine/godot-proposals/issues/2450) [generation filters](https://github.com/godotengine/godot-proposals/issues/9998) [in the future](https://github.com/godotengine/godot-proposals/issues/10667) :slightly_smiling_face:

**Review by akien-mga — APPROVED** [review]
Looks pretty good to me.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**